package.json

```diff
@@ -62,5 +62,14 @@
       "name": "Steve Gill",
       "email": "stevengill97@gmail.com"
     }
-  ]
+  ],
+  "nyc": {
+    "include": [
+      "index.js"
+    ],
+    "reporter": [
+      "lcov",
```

Comment:
Why do we need this?

Reply:
Are you referring to the `lcov` reporter?

It is not exactly required.

I only set this because I had been setting in other areas, e.g. the Electron repo. It was only there to provide alternative visually way of seeing what lines were not covered though generated HTML reports. Developers could just read the line numbers from terminal and look at the code in their editor.

Here is an example:

<img width="1244" alt="cordova-create-missing-coverage" src="https://user-images.githubusercontent.com/1029107/68067033-a4913180-fd84-11e9-8855-ab0e4ee5c230.png">

Reply:
Ah, OK. So we also need `lcov` if we want to setup Codecov for this repo, correct?

Reply:
Yes. Part of the output is needed for codecov and most likely any other code coverage services out there, e.g. coveralls.

* `text` only generates the CLI output.
* `lcov` generates both HTML visual report as well as the raw content in `coverage/lcov.info` file.

Only the raw content file is necessary and shipped to codecov.

The bare minimum process for codecov is actually:

```bash
npx nyc npm t
npx nyc report --reporter=text-lcov > coverage.lcov
npx codecov
```

* Run the tests
* Grab report as lcov raw format and store in `coverage.lcov`
* codecov will magically grab and send the `coverage.lcov`

If we added `text-lcov` instead of `lcov` that will not generate the output file and it will actually print raw content to terminal which will be ugly. << Just making a note here if you ask about this.

Reply:
Cool, thanks for the detailed explanation 🙂